itensor/util/safe_ptr.h

```diff
@@ -18,6 +18,7 @@
 
 #include "itensor/util/print.h"
 #include "itensor/util/stdx.h"
+#include <thrust/device_vector.h>
```

Comment:
Will this include make it remain possible for people to compile ITensor if they aren't going to use CUDA or thrust? Also I'm not familiar with thrust: is it related to GPU or something else?